tests/unit/s2n_client_psk_extension_test.c

```diff
@@ -25,39 +25,33 @@
 #define TEST_BYTES_2 0x0A, 0x0B, 0x0C
 #define TEST_BYTES_SIZE_2 0x00, 0x03
 
-/* identity size + obfuscated_ticket_age_size + (max num of psk identities + 1) */
-#define TOO_MANY_IDENTITIES (sizeof(uint16_t) + sizeof(uint32_t)) * (MAX_NUM_OF_PSK_IDENTITIES + 1)
-
 struct s2n_psk_test_case {
     s2n_hmac_algorithm hmac_alg;
     uint8_t hash_size;
     const uint8_t* identity;
     size_t identity_size;
 };
 
-static int s2n_test_select_psk_identity_callback(struct s2n_connection *conn, struct s2n_psk_identity *identities,
-                                                 size_t identities_length, uint16_t *chosen_wire_index)
+uint16_t s2n_test_customer_wire_index_choice;
+static int s2n_test_select_psk_identity_callback(struct s2n_connection *conn,
+        struct s2n_offered_psk_list *psk_identity_list, uint16_t *chosen_wire_index)
 {
-    notnull_check(conn);
-    notnull_check(identities);
-    notnull_check(chosen_wire_index);
-
-    struct s2n_array *known_psks = &conn->psk_params.psk_list;
-
-    for (size_t i = 0; i < identities_length; i++) {
-        struct s2n_blob wire_identity = { 0 };
-        GUARD(s2n_blob_init(&wire_identity, identities[i].data, identities[i].length));
-
-        struct s2n_psk *local_match = NULL;
-        GUARD_AS_POSIX(s2n_match_psk_identity(known_psks, &wire_identity, &local_match));
+    *chosen_wire_index = s2n_test_customer_wire_index_choice;
+    return S2N_SUCCESS;
+}
 
-        if (local_match != NULL) {
-            *chosen_wire_index = i;
-            return S2N_SUCCESS;
-        }
-    }
+static int s2n_test_error_select_psk_identity_callback(struct s2n_connection *conn,
+        struct s2n_offered_psk_list *psk_identity_list, uint16_t *chosen_wire_index)
+{
+    S2N_ERROR(S2N_ERR_UNIMPLEMENTED);
```

Comment:
We can only expect S2N_SUCCESS/S2N_FAILURE here as the customer specified callback would not be using S2N error codes.

Reply:
I'm not sure I understand the problem. Our test implementation of the callback does whatever we implement it to do. In this case, it sets a specific error code. Our tests can then check for that error code to verify the expected behavior of this callback being triggered but failing.